### PR TITLE
EB-16795 Add callBack base url

### PR DESCRIPTION
**Pull request by aallier**

### CODE REVIEW

File: Plugin/DotpayDirectPlugin.php [hunk 1]
```diff
@@ -497,4 +499,21 @@ protected function getReturnUrl(ExtendedDataInterface $data)
 
         throw new \RuntimeException('You must configure a return url.');
     }
+
+    protected function getCallBackUrl(PaymentInstructionInterface $instruction, $route, $baseUrl): string
+    {
+        if ($baseUrl) {
+            return $this->router->generate(
```
Comment (ch3ric) [inline Plugin/DotpayDirectPlugin.php:506]:
```suggestion
    protected function getCallBackUrl(PaymentInstructionInterface $instruction, $route): string
    {
        if ($this->callBackUrl) {
            return $this->callBackUrl.$this->router->generate(
```

File: Plugin/DotpayDirectPlugin.php [hunk 2]
```diff
@@ -161,11 +167,7 @@ public function createDotpayRedirectActionException(FinancialTransactionInterfac
         $instruction = $transaction->getPayment()->getPaymentInstruction();
 
         $extendedData = $transaction->getExtendedData();
-        $urlc         = $this->router->generate(
-            'ets_payment_dotpay_callback_urlc',
-            ['id' => $instruction->getId()],
-            true
-        );
+        $urlc = $this->getCallBackUrl($instruction, 'ets_payment_dotpay_callback_urlc', $this->callBackUrl);
```
Comment (ch3ric) [inline Plugin/DotpayDirectPlugin.php:170]:
```suggestion
        $urlc = $this->getCallBackUrl($instruction, 'ets_payment_dotpay_callback_urlc');
```

File: Plugin/DotpayDirectPlugin.php [hunk 3]
```diff
@@ -497,4 +499,21 @@ protected function getReturnUrl(ExtendedDataInterface $data)
 
         throw new \RuntimeException('You must configure a return url.');
     }
+
+    protected function getCallBackUrl(PaymentInstructionInterface $instruction, $route, $baseUrl): string
```
Comment (graillus) [inline Plugin/DotpayDirectPlugin.php:503]: Why is this method protected ?
```suggestion
    private function getCallBackUrl(PaymentInstructionInterface $instruction, $route, $baseUrl): string
```

**Review by graillus — APPROVED** [review]

File: Plugin/DotpayDirectPlugin.php [hunk 4]
```diff
@@ -101,6 +104,7 @@ class DotpayDirectPlugin extends AbstractPlugin
      * @param bool              $recipientChk
      * @param bool              $onlineTransfer
      * @param int               $expirationTime
+     * @param string            callback_base_url
```
Comment (ch3ric) [inline Plugin/DotpayDirectPlugin.php:107]:
```suggestion
     * @param string            $callBackUrl
```

File: Plugin/DotpayDirectPlugin.php [hunk 5]
```diff
@@ -90,6 +90,9 @@ class DotpayDirectPlugin extends AbstractPlugin
     /** @var bool */
     protected $expirationTime;
 
+    /** @var bool */
```
Comment (ch3ric) [inline Plugin/DotpayDirectPlugin.php:93]: it's a nullable string, based on what you've set in `Configuration`
```suggestion
    /** @var string|null */
```

File: Plugin/DotpayDirectPlugin.php [hunk 6]
```diff
@@ -112,7 +116,8 @@ public function __construct(
         $chk = false,
         $recipientChk = false,
         $onlineTransfer = false,
-        $expirationTime = 0
+        $expirationTime = 0,
+        $callBackUrl = ''
```
Comment (ch3ric) [inline Plugin/DotpayDirectPlugin.php:120]:
```suggestion
        $callBackUrl = null
```

**Review by ch3ric — APPROVED** [review]

**Review by ch3ric — APPROVED** [review]

**Pull request merged**